### PR TITLE
feat: landing & config page UX improvements

### LINKED ISSUE #58: Improve UX of the root domain landing page
We should use the opportunity before the SW is installed to let the user now how the SW is different from a normal gateway before auto-redirecting.

```[tasklist]
### Tasks
- [ ] Show default configuration in config, i.e. delegated routing, DoH, and trustless gateway endpoints,
- [ ] Inform user about in-browser verification
- [ ] Mention Helia with link to repo
- [ ] Add link to GH repo for issues
- [ ] "Alpha" messaging
- [x] Add a comment at the top of the HTML for when users try URLs with CURL
```

### PR DESCRIPTION
**Pull request by SgtPooki**
## Title

feat: landing & config page UX improvements

## Description

Fixes https://github.com/ipfs-shipyard/service-worker-gateway/issues/58

### Summary of changes

1. Add explainer with links to helper-ui (landing page)
2. Allow setting dnsResolvers in config.
3. Shows default urls in config
4. Requires at least 1 dns resolver/router/gateway
5. Adds "reset" button so users can get back to defaults if they bork things (~~TODO: need to fix this so it doesn't require a refresh, which will not work on first-hit for subdomain~~ this todo is done.)

### Landing page view

image

### Config page view

image

## Notes & open questions

FYI that inputs are still JSON based for now.

## Change checklist

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation if necessary (this includes comments as well)
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by SgtPooki — COMMENTED** [review]
self review.. i'm going to try to provide a config-provider so we can call reset/save/etc there and remove the localStorage items

File: src/index.tsx [hunk 1]
```diff
@@ -31,6 +31,5 @@ root.render(
     <RouterProvider routes={routes}>
       <App />
     </RouterProvider>
-  </React.StrictMode>,
-  container
```
Comment (2color) [inline src/index.tsx:35]: Is this intentional?

Comment (SgtPooki) [reply src/index.tsx:35]: yep! the second param is not needed for react-dom, but was for preact.

**Review by lidel — APPROVED** [review]
Thanks! Content lgtm, but it felt a bit distracting, so played a bit and ended up moving it to bottom.

Pushed a commit with some additional changes:

- moved About to the bottom and made it more friendly for someone who wants to learn how things work
- adjusted colors to improve contrast and indicate when button is enabled
- tweaked the header to include (beta) and match style of other web utiities we have like https://explore.ipld.io/ or https://cid.ipfs.tech/

@SgtPooki @2color  If these look good, feel free to merge or rephrase if anything sounds off.

### Preview

> Screen Shot 2024-05-14 at 22 13 49

**Comment by SgtPooki** [comment]
@lidel that looks a lot cleaner. i'll fix up the CI and get this merged.

**Pull request merged**